students/morgan/session03/mailroom.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+# name, donation total, number of donations and average of donations
+
+
+def main_loop():
+
+    while True:
+        answer = str(input("Select from one of these options:\n"
+              "(1) Send a Thank You\n"
+              "(2) Create a Report\n"
+              "(3) quit\n"
+              "> "))
+        if answer =='3':
+            break
+        elif answer =='1':
+            thank_you()
+        elif answer =='2':
+            print_report(donor_list)
+        else:
+            print("\nPlease type 1, 2, or 3")
+            
+
+    
+
+def print_report(donors):
+    '''need the list of donors sorted before I can print them accurately,
+    found the sorted function while looking for a way to do this'''
+    donors = sorted(donors, key=lambda x: x[1], reverse=True)
+
+    print("\n\n")
+    print("{msg1: <50}|{msg2:<13}|{msg3:<14}|{msg4:<12}".format(msg1='Donor Name',
+        msg2='Total Donated', msg3='Donation Count', msg4='Average Gift'))
+
+    for x in donors:
+        print('{:<50}|{:>12.2f}|{:^14}|{:>12.2f}'.format(x[0],x[1],x[2],x[1]/x[2]))
+
+    print("\n\n")
+
+
+def thank_you():
+    '''takes in the name or 'list' command'''
+    donor_input = input("Enter a donor's full name or type 'list' for all donors \n"
+        "> ")
+    
+    '''if 'list' is chosen than the current list of donor names are printed,
+    no order enforced'''
+    if str.lower(donor_input) == 'list':
+        print('List of donors: \n')
+        for row in donor_list:
+            print(row[0]+'\n')
+   
+    elif name_in_list(donor_input) is False:
+        '''if the donor is not on the current list, it will be added to the list
+    with the donation amount, the number of  donations set to 1 and a
+    thank you leter generated. donation converted to float for safety'''
+        donor_donation = float(input('Enter donation amount \n> '))
+        add_donor(donor_input, donor_donation)
+        write_letter(donor_input, donor_donation)
+
+    elif name_in_list(donor_input):
+        '''if the donor exists in the list, will fetch the index of that entry
+        and add the donation to the donors list and increment the number of donations.
+        I wanted to make sure the correctly capitalized name was kept and used
+        donation converted to float for safety'''
+        proper_name = donor_list[name_in_list(donor_input)][0]
+        donor_donation = float(input('Enter donation amout \n> '))
+        add_donation(proper_name, donor_donation)
+        write_letter(proper_name, donor_donation)
+
```

Comment:
So, usually the reason that you would ask to see all of the donors, is so that you are sure you are writing the name correctly, so you would expect that after asking for the names, you would be prompted again for the thank you note. So, you might want to consider putting in a loop and including exiting to main menu as one of the choices. Once you do this, you may find you can pull stuff in the if blocks into separate functions. 
Notice also that you have typed the line `donor_donation = float(input... ` and the line write_letter twice. DRY (don't repeat yourself) is a principle we will be talking more about...